pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -531,11 +617,13 @@ func isSkipped(rprt *ResolvedPipelineRunTask, stateMap map[string]*ResolvedPipel
 
 	// Recursively look at parent tasks to see if they have been skipped,
 	// if any of the parents have been skipped, skip as well
-	node := d.Nodes[rprt.PipelineTask.Name]
-	for _, p := range node.Prev {
-		skip := isSkipped(stateMap[p.Task.HashKey()], stateMap, d)
-		if skip {
-			return true
+	// continue if the task does not belong to the specified Graph
+	if node, ok := d.Nodes[rprt.PipelineTask.Name]; ok {
+		for _, p := range node.Prev {
+			skip := isSkipped(stateMap[p.Task.HashKey()], stateMap, d)
+			if skip {
+				return true
+			}
```

Comment:
im confused why this changed - is it a bug? if so, maybe it could get fixed in another PR? (or there is something finally related that im missing)

Reply:
nope, this is finally related, git diff is not helpful at all 😢  the only change here is:

```
+ 	if node, ok := d.Nodes[rprt.PipelineTask.Name]; ok {
```

this additional check is needed because now, finally tasks are also part of same `pipelineState` which used to only hold DAG tasks. This check avoids the `for` loop if `node` does not exist in the specified `Graph`. Otherwise node will be `nil` in case its not part of that `Graph` and results in calling `Prev` on `nil` (get all parents).

Reply:
yeah since we are creating two graphs with this PR, one for DAG tasks and [one for finally tasks](/test pull-tekton-pipeline-integration-tests) and`pipelineState` holds resolved tasks from both sections. 

Yes this will be much clear after @afrittoli and I are done with refactoring.

Reply:
NIT: perhaps it would be slightly more readable if we had a function for that, say `isPipelineTaskinDag`... or we just wait for a refactor on top.

Reply:
We have multiple choices `state.isPipelineTaskinDag`, `ResolvedPipelineRunTask.InDag()` etc,  added this in the issue #2852